pkg/skaffold/deploy/helm.go

```diff
@@ -308,18 +310,37 @@ func (h *HelmDeployer) deployRelease(ctx context.Context, out io.Writer, r lates
 		return nil, fmt.Errorf("release args: %w", err)
 	}
 
-	iErr := h.exec(ctx, out, r.UseHelmSecrets, args...)
-
-	var b bytes.Buffer
+	err = h.exec(ctx, out, r.UseHelmSecrets, args...)
+	if err != nil {
+		return nil, fmt.Errorf("install: %w", err)
+	}
 
-	// Be accepting of failure
-	if err := h.exec(ctx, &b, false, getArgs(helmVersion, releaseName, opts.namespace)...); err != nil {
-		logrus.Warnf(err.Error())
-		return nil, nil
+	b, err := h.getRelease(ctx, helmVersion, releaseName, opts.namespace)
+	if err != nil {
+		return nil, fmt.Errorf("get release: %w", err)
 	}
 
 	artifacts := parseReleaseInfo(opts.namespace, bufio.NewReader(&b))
-	return artifacts, iErr
+	return artifacts, nil
+}
+
+// getRelease confirms that a release is visible to helm
+func (h *HelmDeployer) getRelease(ctx context.Context, helmVersion semver.Version, releaseName string, namespace string) (bytes.Buffer, error) {
+	// Retry, because under Helm 2, at least, a release may not be immediately visible
+	opts := backoff.NewExponentialBackOff()
+	opts.MaxElapsedTime = 4 * time.Second
+	var b bytes.Buffer
+
+	err := backoff.Retry(
+		func() error {
+			if err := h.exec(ctx, &b, false, getArgs(helmVersion, releaseName, namespace)...); err != nil {
+				logrus.Debugf("unable to get release: %v (may retry):\n%s", err, b.String())
```

Comment:
Can you use `backoff.RetryNotify()` and move the retry notice to the Notify function? (Saying _will retry in N seconds_ seems useful.)

Reply:
Honestly, since it's just a debug line that likely no one will ever see, I'm just going to leave it like this until it becomes a common enough to complicate things for a better message.

Thanks for the heads up on backoff.RetryNotify!